src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/advice/ApplicationProblemControllerAdviceTest.java

```diff
@@ -429,5 +459,19 @@ private static class GenericExceptionScenario {
         Status expectedStatus;
     }
 
+    public static <T> T safeCast(Object object, Class<T> clazz) {
+        if (clazz.isInstance(object)) {
+            return clazz.cast(object);
+        } else {
+            log.error("Expected object of type {} but received: {}",
+                         clazz.getName(),
+                         object != null ? object.getClass().getName() : "null");
+
+            throw new IllegalArgumentException(
+                "Object is not of type " + clazz.getName() + "."
+            );
+        }
+    }
```

Comment:
Is there value in this since it is a test class anyway.  If the cast is invalid it ought to throw a ClassCastException exception anyway?

Reply:
This was added just for the testing context Ravi.